### ISSUE TITLE
Fix `declaration-property-value-keyword-no-deprecated` false negatives for non-color functions

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#9161]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
  /* background as color in attr */
  color: attr( foo type(<color>), background);
}

a {
  /* background as color */
  color: var(--foo, background);
}

a {
  /* background as color */
  color: rgb(from var(--foo, background) r g b);
}

a {
  /* background as color */
  color: rgb(from if(style(--foo: bar): background; else: white) r g b);
}

a {
  /* background as a random keyword */
  color: --foo(background);
}

a {
  /* background as a random keyword */
  color: if(style(--foo: background): red; else: white);
}
```

Ref https://github.com/stylelint/stylelint/pull/9116#pullrequestreview-3969030324

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  rules: {
    "declaration-property-value-keyword-no-deprecated":true
  }
};
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilB6AVCgEZJgDWA5gE4QCumUKSAzipADYRUoCW2SAFwHd8uLDg5d4TIVQAUKAGYQIKAQE8ADjDkAeSVQB8ASgA0xUpRr0oxgNxYAvliyoM2PIRLlqdBk1YDFFFxNghOKmkANyR5AFo45Qhzbys-WwdMZ0xXNFCCCx9rfxYwiOCxDwNpKgoiOUUaAFsUGPjElRTLXxtjFG4KYnsnF0w3fK9u4sZSoJCq8Kl+uobm3kU5Zg12HQSk6RIqYwOp9LsUGHZmGGkAdwALHgEYPoGhzOzc9xwC1J6S1ioKhIBgQFpkGDqW5cRjzCSLSIoPYqOR-abDLKjcYeX6nGwBJj9EFQMEoCFQmEVULVdabba7DoQE5FdLHfowKDnS7XO6PZ4Y7IgUwgRQ8HYAOSQTTgiBgAA8pZodgA6MDMZhC8AQTCiigIEDyzRcAQoWCKJC0dgm779S0wZjSG04dAgWBgdixQQ8bVxTQ0bRUDRxGLsWgwOLk6FUKBxTAQOKwP0wMCCDku+DCMOhbKOTKayA6ngUABiXCagn1W3UO3YfAEqu1uuVTQAVhrhYmNYgbS6qzW6+mUC6Pc8ti7TKFe-Ta5gBHEC7q4ltibEoIPh6mxyAnCBHEA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "17.5.0",
}
```

### What did you expect to happen?

Problems to be reported.

### What actually happened?

No problems reported.

### Do you have a proposal to fix the bug?

See discussions in https://github.com/stylelint/stylelint/pull/8579#discussion_r2094065268 and https://github.com/stylelint/stylelint/issues/8575#issuecomment-3694752407.

**Comment by hriztam** [comment]
This rule is missing deprecated color keywords like `background` when they appear inside newer color-related patterns such as `attr(<color>, background)`, `var()` fallbacks, and `rgb(from …)`. I’m planning to teach the rule to treat those as color contexts and report them, while still ignoring cases where `background` is just a generic identifier or function name like `--foo(background)`.

As I also solved the #9104 issue, I think I have an idea should I implement this
